subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -122,76 +122,94 @@ func (h *HivedAlgorithm) Schedule(pod *core.Pod, suggestedNodes []string) intern
 		groupPhysicalPlacement, groupVirtualPlacement, s.GpuNumber, podIndex, newGroup, suggestedNodes, pod)
 }
 
-// TODO: reconfig; suggestedNodes (failure)
+// TODO: suggestedNodes (failure)
 func (h *HivedAlgorithm) AddAllocatedPod(pod *core.Pod) {
 	h.algorithmLock.Lock()
 	defer h.algorithmLock.Unlock()
 
-	klog.Infof("[%v]: adding allocated pod...", internal.Key(pod))
 	s := internal.ExtractPodSchedulingSpec(pod)
 	info := internal.ExtractPodBindInfo(pod)
+	klog.Infof("[%v]: adding allocated pod (node %v, GPUs %v)...", internal.Key(pod), info.Node, info.GpuIsolation)
 
 	chain := CellChain(info.CellChain)
 	hasVirtualPlacement := true
+	downgrade := false
 	if group := h.allocatedAffinityGroups[s.AffinityGroup.Name]; group == nil {
 		newGroup := newAlgoAffinityGroup(s.AffinityGroup)
 		for _, gms := range info.AffinityGroupBindInfo {
 			gpuNumber := int32(len(gms.PodPlacements[0].PhysicalGpuIndices))
-			newGroup.physicalGpuPlacement[gpuNumber] = make([]CellList, len(gms.PodPlacements))
-			if hasVirtualPlacement {
-				newGroup.virtualGpuPlacement[gpuNumber] = make([]CellList, len(gms.PodPlacements))
-			}
 			for podIndex := int32(0); podIndex < int32(len(gms.PodPlacements)); podIndex++ {
-				newGroup.physicalGpuPlacement[gpuNumber][podIndex] = make(
-					CellList, len(gms.PodPlacements[podIndex].PhysicalGpuIndices))
-				if hasVirtualPlacement {
-					newGroup.virtualGpuPlacement[gpuNumber][podIndex] = make(
-						CellList, len(gms.PodPlacements[podIndex].PhysicalGpuIndices))
-				}
 				node := gms.PodPlacements[podIndex].PhysicalNode
 				for gpuIndex := int32(0); gpuIndex < int32(
 					len(gms.PodPlacements[podIndex].PhysicalGpuIndices)); gpuIndex++ {
 					physicalGpuIndex := gms.PodPlacements[podIndex].PhysicalGpuIndices[gpuIndex]
-					pGpu := h.findPhysicalGpu(chain, node, physicalGpuIndex)
-					if pGpu == nil {
-						panic(fmt.Sprintf("[%v]: physical GPU cell not found when adding pod: node %v, GPU index %v",
-							internal.Key(pod), node, physicalGpuIndex))
-					}
-					newGroup.physicalGpuPlacement[gpuNumber][podIndex][gpuIndex] = pGpu
-
-					var vGpu *VirtualCell
-					if hasVirtualPlacement {
-						virtualCellIndex := gms.PodPlacements[podIndex].VirtualCellIndices[gpuIndex]
-						if virtualCellIndex >= 0 {
-							vGpu = h.findVirtualGpu(s.VirtualCluster, chain, s.ReservationId, virtualCellIndex)
-							if vGpu == nil {
-								panic(fmt.Sprintf("[%v]: virtual GPU cell not found when adding pod: virtual cell index %v",
-									internal.Key(pod), virtualCellIndex))
+					if pGpu := h.findPhysicalGpu(chain, node, physicalGpuIndex); pGpu == nil {
+						klog.Infof(
+							"[%v]: cannot find GPU %v on node %v due to reconfiguration: deleted from the spec",
```

Comment:
>reconfiguration [](start = 51, length = 15)

you only know inconsistent, it may not be caused by reconfiguration, remove the word "reconfiguration" and check other places/

Reply:
how to check? only if the scheduler knows it is during reconfiguration

---
In reply to: [339986481](https://github.com/microsoft/pai/pull/3768#discussion_r339986481) [](ancestors = 339986481)